roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -180,62 +177,12 @@ impl ParseClientJobDeclarationMessages for JobDeclaratorDownstream {
         }
     }
 
-    fn handle_submit_solution(&mut self, message: SubmitSolutionJd) -> Result<SendTo, Error> {
-        //TODO: implement logic for success or error
-        let (last_declare, mut tx_list, _) = match self.declared_mining_job.take() {
-            Some((last_declare, tx_list, _x)) => (last_declare, tx_list, _x),
-            None => {
-                warn!("Received solution but no job available");
-                return Ok(SendTo::None(None));
-            }
-        };
-        let coinbase_pre = last_declare.coinbase_prefix.to_vec();
-        let extranonce = message.extranonce.to_vec();
-        let coinbase_suf = last_declare.coinbase_suffix.to_vec();
-        let mut path: Vec<Vec<u8>> = vec![];
-        for tx in &tx_list {
-            let id = tx.txid();
-            let id = id.as_ref().to_vec();
-            path.push(id);
-        }
-        let merkle_root =
-            merkle_root_from_path(&coinbase_pre[..], &coinbase_suf[..], &extranonce[..], &path)
-                .expect("Invalid coinbase");
-        let merkle_root = Hash::from_inner(merkle_root.try_into().unwrap());
-
-        let prev_blockhash = u256_to_block_hash(message.prev_hash.into_static());
-        let header = stratum_common::bitcoin::blockdata::block::BlockHeader {
-            version: last_declare.version as i32,
-            prev_blockhash,
-            merkle_root,
-            time: message.ntime,
-            bits: message.nbits,
-            nonce: message.nonce,
-        };
-
-        let coinbase = [coinbase_pre, extranonce, coinbase_suf].concat();
-        let coinbase = Transaction::deserialize(&coinbase[..]).unwrap();
-        tx_list.insert(0, coinbase);
-
-        let mut block = Block {
-            header,
-            txdata: tx_list.clone(),
-        };
-
-        block.header.merkle_root = block.compute_merkle_root().unwrap();
-
-        let serialized_block = serialize(&block);
-        let hexdata = hex::encode(serialized_block);
-
-        // TODO This line blok everything!!
-        self.mempool
-            .safe_lock(|x| {
-                if let Some(client) = x.get_client() {
-                    client.submit_block(hexdata).unwrap();
-                }
-            })
-            .unwrap();
+    fn handle_submit_solution(
+        _self: Arc<Mutex<Self>>,
+        message: SubmitSolutionJd<'_>,
+    ) -> Result<SendTo, Error> {
+        let m = JobDeclaration::SubmitSolution(message.clone().into_static());
 
-        Ok(SendTo::None(None))
+        Ok(SendTo::RelayNewMessage(m))
```

Comment:
not sure if this is semantically correct RelayNewMessage(m) meand relay a new sv2 message and the new message is the one contained inside (m) but this is not what you want to do

Reply:
This should be rather be SendTo::None(m)

Reply:
using relay new message with the same message received do not make much sense to me. It should be realy same message but this is not the case

Reply:
Now the the handler returns a SendTo(None(Some(JobDeclaration::SubmitSolution())))